File: Golang/parallelchange/shopping_cart_test.go
package parallelchange

import "testing"

func TestShoppingCartMayHaveJustOneItem(t *testing.T) {
	shoppingCart := NewShoppingCart()
	shoppingCart.Add(10)

	expected := 1
	if actual := shoppingCart.NumberOfProducts(); actual != expected {
		t.Errorf("Expected number of products to be %d, but got %d", expected, actual)
	}
}

func TestShoppingCartHasTotalPriceEqualToTotalPriceOfItsContents(t *testing.T) {
	shoppingCart := NewShoppingCart()
	shoppingCart.Add(10)

	expected := 10
	if actual := shoppingCart.CalculateTotalPrice(); actual != expected {
		t.Errorf("Expected total price to be %d, but got %d", expected, actual)
	}
}

func TestShoppingCartHasDiscountWhenContainsAtLeastOnePremiumItem(t *testing.T) {
	shoppingCart := NewShoppingCart()
	shoppingCart.Add(100)

	expected := true
	if actual := shoppingCart.HasDiscount(); actual != expected {
		t.Errorf("Expected 'has discount' to be %v, but got %v", expected, actual)
	}
}

func TestShoppingCartDoesNotHaveDiscountWhenAllItemsAreCheap(t *testing.T) {
	shoppingCart := NewShoppingCart()
	shoppingCart.Add(10)

	expected := false
	if actual := shoppingCart.HasDiscount(); actual != expected {
		t.Errorf("Expected 'has discount' to be %v, but got %v", expected, actual)
	}
}

File: PHP/tests/ShoppingCartTest.php
<?php
declare(strict_types=1);

use ParallelChange\ShoppingCart;

class ShoppingCartTest extends PHPUnit_Framework_TestCase {
	public function testShoppingCartMayHaveJustOneItem() {
		$shoppingCart = new ShoppingCart();
		$shoppingCart->add(10);

		$this->assertSame(1, $shoppingCart->getNumberOfProducts());
	}

	public function testShoppingCartHasTotalPriceEqualToTotalPriceOfItsContents() {
		$shoppingCart = new ShoppingCart();
		$shoppingCart->add(10);

		$this->assertSame(10, $shoppingCart->calculateTotalPrice());
	}

	public function testShoppingCartHasDiscountWhenContainsAtLeastOnePremiumItem() {
		$shoppingCart = new ShoppingCart();
		$shoppingCart->add(100);

		$this->assertTrue($shoppingCart->hasDiscount());
	}

	public function testShoppingCartDoesNotHaveDiscountWhenAllItemsAreCheap() {
		$shoppingCart = new ShoppingCart();
		$shoppingCart->add(10);

		$this->assertFalse($shoppingCart->hasDiscount());
	}
}
